nemoguardrails/colang/v2_x/library/avatars.co

```diff
@@ -274,11 +283,14 @@ flow managing thinking posture
   match UtteranceUserAction.Finished()
   if not $bot_talking_state
     start bot posture "thinking"
-    bot started saying something
-
-@loop("managing_idle_posture")
-flow managing idle posture
-  """Start and stop idle posture."""
-  match (polling llm request response).Started()
-  start bot posture "Thinking, idle" as $posture
-  match (polling llm request response).Finished()
+    if not $bot_talking_state
```

Comment:
This seems to unnecessary since we already check before?

Reply:
I think the issue here is that since this in a separate loop you can have the StartPostureBotAction() at the "same time" as a StartUtteranceBotAction() from a different loop. (So basically, the value of $bot_talking_state changes during the line 285 so we check again on 286).

Reply:
Good point... That seems to call for an additional Colang feature imo. Something to define a scope where a certain state must hold...

Reply:
Yes I agree. We should think about that.